app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHours.kt

```diff
@@ -93,12 +93,25 @@ class AddOpeningHours(
             ),
         ).map { it.key + " ~ " + it.value.joinToString("|") }.joinToString("\n or ") + "\n" + """
             )
-            and !opening_hours
+            and (!opening_hours or opening_hours older today -1 years)
+          )
+          or
+          (
+          opening_hours older today -1 years
+          and
+            (
+            leisure=park
+            or
+            barrier
+            or
+            amenity=toilets
+            or
+            amenity=bicycle_rental
+            )
           )
-          or opening_hours older today -1 years
         )
         and access !~ private|no
-        and (name or brand or noname = yes or name:signed = no or amenity=recycling)
+        and (name or brand or noname = yes or name:signed = no or amenity=recycling or amenity=toilets)
```

Comment:
```suggestion
        and (
          name or brand or noname = yes or name:signed = no
          or amenity = recycling or amenity = toilets or barrier
        )
```

too long line, formatting, barriers also usually have no name